src/fhirtypes/common.ts

```diff
@@ -301,8 +321,17 @@ export function replaceReferences<T extends AssignmentRule | CaretValueRule>(
   } else if (value instanceof FshCode) {
     const [system, ...versionParts] = value.system?.split('|') ?? [];
     const version = versionParts.join('|');
-    const codeSystem = tank.fish(system, Type.CodeSystem);
-    const codeSystemMeta = fisher.fishForMetadata(codeSystem?.name, Type.CodeSystem);
+    let codeSystem = tank.fish(system, Type.CodeSystem);
```

Comment:
Again going to appeal to the higher power of @cmoesel here, creator of the issue. I was thinking that something like
```
Instance: FooBar
InstanceOf: CodeSystem
Usage: #definition
* name = "FooBar"
* title = "Foo Bar"
* url = "http://codesystem.com/foobar"
* status = #active
* content = #complete
* concept[+].code = #foo
* concept[+].code = #bar

ValueSet: V
* include codes from system FooBar
```
should also work, but currently it does not. The issue was specific resolving FSH codes, but it feels a little confusing to me to support that case and not this one. I'm wondering if it would be possible to make this happen inside the `tank` itself. What if when you fished for a `CodeSystem`, we looked through `CodeSystems` first, but then looked through any `Instances` with `InstanceOf: CodeSystem` set. Definitely this could be not possible for some reason I'm not seeing, but what do you think?

Reply:
It's definitely possible. I had considered implementing this from the `fish` method instead of in `replaceReferences`, but wasn't sure if there was some reason we wouldn't want this to happen in all cases so I went for the slightly narrower option. I do agree that facially it makes sense for these instance-defined CodeSystems to resolve in all areas that a regular CodeSystem would.

Reply:
Commit `081fd08` moves this functionality into the tank so this fix is a bit more wide-ranging... I don't see any obvious problems with this approach but if @cmoesel objects then it can be rolled back.